### ISSUE TITLE
Object Parameter with Anonymous Record

### ISSUE THREAD

**Issue by sayurin** [dotnet/fsharp#6395]
Following code causes compile error.
```
System.Console.WriteLine {| x = 1 |}
```

Error message is:
> parameter error FS0192: error : the anonymous record <>f__AnonymousType557104394`1' has not been generated in the pre-phase of generating this module

Compiler version is:
> Microsoft (R) F# Compiler version 10.4.0 for F# 4.6

**Comment by dsyme** [comment]
Odd, can't repro this on master
```
C:\GitHub\dsyme\visualfsharp2>artifacts\bin\fsc\Debug\net46\fsc --optimize- a.fs
Microsoft (R) F# Compiler version 10.2.3 for F# 4.5
Copyright (c) Microsoft Corporation. All Rights Reserved.
```

**Comment by sayurin** [comment]
I'm using Visual Studio 2019 RC, 16.0.0 RC.4.

In Developper Command Prompt, following stack traces sre dumped.

```
C:\Users\sayurin\Desktop>fsc --optimize- a.fs
Microsoft (R) F# Compiler version 10.4.0 for F# 4.6
Copyright (c) Microsoft Corporation. All Rights Reserved.

error FS0192: internal error: the anonymous record <>f__AnonymousType320777345`1' has not been generated in the pre-phase of generating this module

Unhandled Exception: System.Exception: the anonymous record <>f__AnonymousType320777345`1' has not been generated in the pre-phase of generating this module
   at Microsoft.FSharp.Core.PrintfModule.PrintFormatToStringThenFail@1647.Invoke(String message)
   at Microsoft.FSharp.Compiler.IlxGen.GenAllocAnonRecd(cenv cenv, CodeGenBuffer cgbuf, IlxGenEnv eenv, AnonRecdTypeInfo anonInfo, FSharpList`1 tyargs, FSharpList`1 args, range m, sequel sequel)
   at Microsoft.FSharp.Compiler.IlxGen.GenExpr(cenv cenv, CodeGenBuffer cgbuf, IlxGenEnv eenv, EmitSequencePointState sp, Expr expr, sequel sequel)
   at Microsoft.FSharp.Compiler.IlxGen.GenExprs@2143.Invoke(Expr e)
   at Microsoft.FSharp.Primitives.Basics.List.iter[T](FSharpFunc`2 f, FSharpList`1 x)
   at Microsoft.FSharp.Compiler.IlxGen.GenILCall(cenv cenv, CodeGenBuffer cgbuf, IlxGenEnv eenv, Boolean virt, Boolean valu, Boolean newobj, ValUseFlag valUseFlags, Boolean isDllImport, ILMethodRef ilMethRef, FSharpList`1 enclArgTys, FSharpList`1 methArgTys, FSharpList`1 argExprs, FSharpList`1 returnTys, range m, sequel sequel)
   at Microsoft.FSharp.Compiler.IlxGen.GenExpr(cenv cenv, CodeGenBuffer cgbuf, IlxGenEnv eenv, EmitSequencePointState sp, Expr expr, sequel sequel)
   at Microsoft.FSharp.Compiler.IlxGen.GenModuleDef@6061-3.Invoke(ModuleOrNamespaceExpr x)
   at Microsoft.FSharp.Primitives.Basics.List.iter[T](FSharpFunc`2 f, FSharpList`1 x)
   at Microsoft.FSharp.Compiler.IlxGen.GenModuleBinding(cenv cenv, CodeGenBuffer cgbuf, QualifiedNameOfFile qname, List`1 lazyInitInfo, IlxGenEnv eenv, range m, ModuleOrNamespaceBinding x)
   at Microsoft.FSharp.Compiler.IlxGen.GenModuleDef@6049-1.Invoke(ModuleOrNamespaceBinding x)
   at Microsoft.FSharp.Primitives.Basics.List.iter[T](FSharpFunc`2 f, FSharpList`1 x)
   at Microsoft.FSharp.Compiler.IlxGen.GenModuleDef@6061-3.Invoke(ModuleOrNamespaceExpr x)
   at Microsoft.FSharp.Primitives.Basics.List.iter[T](FSharpFunc`2 f, FSharpList`1 x)
   at Microsoft.FSharp.Compiler.IlxGen.GenModuleDef@6061-3.Invoke(ModuleOrNamespaceExpr x)
   at Microsoft.FSharp.Primitives.Basics.List.iter[T](FSharpFunc`2 f, FSharpList`1 x)
   at Microsoft.FSharp.Compiler.IlxGen.GenModuleExpr@6030.Invoke(Tuple`2 scopeMarks)
   at Microsoft.FSharp.Compiler.IlxGen.LocalScope[a](String nm, CodeGenBuffer cgbuf, FSharpFunc`2 f)
   at Microsoft.FSharp.Compiler.IlxGen.GenTopImpl@6155-1.Invoke(CodeGenBuffer cgbuf, IlxGenEnv eenv)
   at Microsoft.FSharp.Compiler.IlxGen.CodeGenThen(cenv cenv, AssemblyBuilder mgbuf, FSharpList`1 entryPointInfo, String methodName, IlxGenEnv eenv, Int32 alreadyUsedArgs, FSharpFunc`2 codeGenFunction, range m)
   at Microsoft.FSharp.Compiler.IlxGen.CodeGenMethod(cenv cenv, AssemblyBuilder mgbuf, FSharpList`1 entryPointInfo, String methodName, IlxGenEnv eenv, Int32 alreadyUsedArgs, FSharpFunc`2 codeGenFunction, range m)
   at Microsoft.FSharp.Compiler.IlxGen.GenTopImpl(cenv cenv, AssemblyBuilder mgbuf, FSharpOption`1 mainInfoOpt, IlxGenEnv eenv, TypedImplFile _arg11, FSharpFunc`2 optimizeDuringCodeGen)
   at Microsoft.FSharp.Compiler.IlxGen.CodegenAssembly(cenv cenv, IlxGenEnv eenv, AssemblyBuilder mgbuf, FSharpList`1 fileImpls)
   at Microsoft.FSharp.Compiler.IlxGen.GenerateCode(cenv cenv, AnonTypeGenerationTable anonTypeTable, IlxGenEnv eenv, TypedAssemblyAfterOptimization _arg1, FSharpList`1 assemAttribs, FSharpList`1 moduleAttribs)
   at Microsoft.FSharp.Compiler.IlxGen.IlxAssemblyGenerator.GenerateCode(IlxGenOptions codeGenOpts, TypedAssemblyAfterOptimization typedAssembly, FSharpList`1 assemAttribs, FSharpList`1 moduleAttribs)
   at Microsoft.FSharp.Compiler.Driver.main2b[a,b,c,d,e,f,g,h](FSharpOption`1 tcImportsCapture, FSharpOption`1 dynamicAssemblyCreator, Args`1 _arg1)
   at Microsoft.FSharp.Compiler.Driver.typecheckAndCompile(CompilationThreadToken ctok, String[] argv, Resolver legacyReferenceResolver, Boolean bannerAlreadyPrinted, ReduceMemoryFlag reduceMemoryUsage, CopyFSharpCoreFlag defaultCopyFSharpCore, Exiter exiter, ErrorLoggerProvider loggerProvider, FSharpOption`1 tcImportsCapture, FSharpOption`1 dynamicAssemblyCreator)
   at Microsoft.FSharp.Compiler.CommandLineMain.Driver.main(String[] argv)
   at Microsoft.FSharp.Compiler.CommandLineMain.main(String[] argv)
   at Microsoft.FSharp.Compiler.ErrorLogger.ErrorLoggerExtensions.ReraiseIfWatsonable(Exception exn)
   at Microsoft.FSharp.Compiler.ErrorLogger.ErrorLoggerExtensions.ErrorLogger.ErrorRecovery(ErrorLogger x, Exception exn, range m)
   at Microsoft.FSharp.Compiler.ErrorLogger.errorRecovery(Exception exn, range m)
   at Microsoft.FSharp.Compiler.CommandLineMain.main(String[] argv)
```

**Comment by cartermp** [comment]
Can repro on the RC. @dsyme Can you think of any meaningful diff between master and dev16.0 w.r.t. this issue?

**Comment by cartermp** [comment]
I think this may also be a bug in the desktop compiler; using this with F# 4.6 on .NET Core 3 preview and `dotnet run` succeeds.

**Comment by kevmal** [comment]
I was hitting this error while using ML.NET static pipelines in fsi. Would often need to reset fsi to get things to work.

**Comment by dsyme** [comment]
No, I can only think it's a mistake in integration

**Comment by dsyme** [comment]
@cartermp Can you check again? I think this was fixed by the fixes yesterday

**Comment by cartermp** [comment]
I can check against the nightlies when I'm in office. But it should definitely work given this overload is the one being used: https://referencesource.microsoft.com/#mscorlib/system/console.cs,2075

**Comment by cartermp** [comment]
Confirmed as fixed in 16.2.
